cvat/apps/engine/media_extractors.py

```diff
@@ -89,17 +89,19 @@ def __init__(self, source_path, dest_path, image_quality, step=1, start=0, stop=
         )
 
         self._dimensions = []
-        file_ = convert_from_path(self._source_path)
-        self._basename = os.path.splitext(os.path.basename(self._source_path))[0]
-        for page_num, page in enumerate(file_):
-            output = os.path.join(self._temp_directory, self._basename + str(page_num) + '.jpg')
-            self._dimensions.append(page.size)
-            page.save(output, 'JPEG')
+        count = 0
+        for source in source_path:
+            pages = convert_from_path(source)
+            for page in pages:
+                output = os.path.join(self._temp_directory, str(count) + '.jpg')
+                count += 1
+                self._dimensions.append(page.size)
+                page.save(output, 'JPEG')
 
         self._length = len(os.listdir(self._temp_directory))
 
     def _get_imagepath(self, k):
-        img_path = os.path.join(self._temp_directory, self._basename + str(k) + '.jpg')
+        img_path = os.path.join(self._temp_directory, str(k) + '.jpg')
```

Comment:
@benhoff ,

How to map annotations of files to original's pdf files?

Reply:
I don't think it's possible with how this code base is setup.

Maybe track the information in the JPG metadata? The extractor could theoretically pull that information out on the backside.

Reply:
@benhoff , previously you have _basename. Thus it was possible to match a jpg with corresponding pdf+page. Also I treat a PDF file as a video file which can contain multiple pages (frames). If somebody needs to annotate multiple PDF files it is necessary to create multiple tasks.

I will vote to be consistent. If we want to allow multiple pdfs per task we need to do the same for video files.

Reply:
@nmanovic , I agree that it would be nice, I'm just not sure how to program it. If you look at the `__iter__` method in the class, that is the problem. By previously having the `_basename` be consistent, and knowing how many files are in the directory, it was possible to easily iterate over all the files.

Is it possible to nest subdirectories? That would be one way to keep the files cleanly separated.

Reply:
@benhoff , let's fix only problem with the docker file and process only one pdf for now. I like the solution with one pdf. Multiple PDFs are inconsistent with archives and video files.